components/cluster.py

```diff
@@ -167,13 +221,13 @@ def get_figures(edges_np, df):
     G = nx.DiGraph()
     G.add_edges_from(edges)
     # this threshold value is experimental and subject to changes.
-    th = 1e3
+    th = 1e-5
     to_remove = []
     # this loop will remove the weak edges. weak edges are the ones
     # with a distance above the threshold  defined above as th.
     for e in G.edges:
         w = G.edges[e]["w"]
-        if w >= th:
+        if 1.0/w >= th:
```

Comment:
This sometimes causes a divide by zero, maybe add a check in case of that?

Reply:
Mmmm it is kind of weird that the w value is zero ( since it is the distance ) It would be zero only for an earthquake compared to itself. I can change it for something like 1.0/( w + 1e-10 ) to avoid the divide by zero.

Reply:
Maybe it was because I was using the Basel data to test the functionality. There, all the first 1000 earthquakes occurres very nearly in the same location, possibly causing the distance to be rounded to zero

Reply:
Adding the small constant to the divisor might be a good idea, as I still get the error every once in a while. Strangely enough, it doesn't seem to actually cause anything apart from a message in the console..

Reply:
Added small constant ( 1e-10)